src/lib/config.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    env,
+    fs::{canonicalize, read_to_string},
+    io::{stderr, Write},
+    path::Path,
+    process,
+};
+
+use getopts::Options;
+
+pub struct Config {
+    /// The maximum number of parallel jobs to run.
+    pub maxjobs: usize,
+    /// The port to listen on.
+    pub port: u16,
+    /// A *fully canonicalised* path to the directory containing per-repo programs.
+    pub reposdir: String,
+    /// The GitHub secret used to validate requests.
+    pub secret: String,
+    /// An optional email address to send errors to.
+    pub email: Option<String>,
+}
+
+impl Config {
+    /// Create a `Config`.
```

Comment:
I'd probably drop that comment. It's obvious even by my standards :)

Reply:
Agreed, but since it's a public(ish) function, it does feel better than nothing.